src/snapred/backend/service/MetadataLookupService.py

```diff
@@ -0,0 +1,44 @@
+from mantid.kernel import IntArrayProperty
+
+from snapred.backend.data.DataFactoryService import DataFactoryService
+from snapred.backend.log.logger import snapredLogger
+from snapred.backend.service.Service import Service
+from snapred.meta.Config import Config
+from snapred.meta.decorators.FromString import FromString
+from snapred.meta.decorators.Singleton import Singleton
+from snapred.meta.validator.RunNumberValidator import RunNumberValidator
+
+logger = snapredLogger.getLogger(__name__)
+
+
+@Singleton
+class MetadataLookupService(Service):
+    dataFactoryService = "DataFactoryService"
+
+    # register the service in ServiceFactory please!
+    def __init__(self):
+        super().__init__()
+        self.dataFactoryService = DataFactoryService()
+        self.registerPath("", self.verifyMultipleRuns)
+        return
+
+    @staticmethod
+    def name():
+        return "verifiedRuns"
+
+    @FromString
+    def verifyMultipleRuns(self, runs: str):
+        maxRuns = Config["instrument.maxNumberOfRuns"]
+        try:
+            allRuns = IntArrayProperty(name="RunList", values=runs).value
+        except Exception as e:
+            raise e
+        validRuns = []
+        for run in allRuns:
+            if RunNumberValidator.validateRunNumber(str(run)):
+                validRuns.append(run)
+        if len(validRuns) > maxRuns:
+            logger.warning(f"Maximum vale of {maxRuns} run numbers exceeded")
+            validRuns = []
```

Comment:
Are you sure you want it to return an empty list?

Reply:
Wouldn't it be better to have the for loop limit to 10, and just return the first ten runs?

Reply:
So I thought about that, but I wasn't sure if it was better to return the first 10 runs or last 10 runs. I decided that if there's more than 10 runs, then the input is wrong, and the output should be empty so that the user has to try again.

Reply:
Would raising a recoverable error work better, then?

Reply:
I think this should? We normally have a box popup whenever we log a warning